src/database_migration.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the DatabaseMigration class to manage database migrations."""
+
+import logging
+from typing import Literal, cast
+
+import ops
+from ops.pebble import ExecError
+
+from constants import FLASK_STATE_DIR
+from exceptions import CharmConfigInvalidError
+
+logger = logging.getLogger(__name__)
+
+
+DatabaseMigrationStatus = Literal["PENDING", "COMPLETED", "FAILED"]
+
+
+class DatabaseMigration:
+    """The DatabaseMigration class that manages database migrations.
+
+    Attrs:
+        script: the database migration script.
+        COMPLETED: one of the possible database migration status.
+        FAILED: one of the possible database migration status.
+        PENDING: one of the possible database migration status.
+    """
+
+    _STATUS_FILE = FLASK_STATE_DIR / "database-migration-status"
+    _COMPLETED_SCRIPT_FILE = FLASK_STATE_DIR / "completed-database-migration"
+
+    COMPLETED: DatabaseMigrationStatus = "COMPLETED"
+    FAILED: DatabaseMigrationStatus = "FAILED"
+    PENDING: DatabaseMigrationStatus = "PENDING"
```

Comment:
As noted elsewhere, these would be better to be associated with the module rather than the class

Reply:
See: https://github.com/canonical/flask-k8s-operator/pull/50#discussion_r1318248094, thanks!